### ISSUE TITLE
make error/warning signal delay configurable

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#211363]
A 1-second delay, in my opinion, does not significantly improve user experience. Ideally, this delay should be configurable, with a default setting of at least 3 seconds. setting `editor.hover.delay` is a viable workaround; however, it falls short in addressing the issue when an error or warning state has already been triggered.

_Originally posted by @rperez030 in https://github.com/microsoft/vscode/issues/204257#issuecomment-2072394714_

**Comment by meganrogge** [comment]
https://github.com/microsoft/vscode/commit/cedd2f4286b5f2547c3a36cfa7899024997eee3c#diff-f62c4db7e6acf19e65e9fe2a80ecc1200cd3412758a69a17e3c268ac6817f90dR107

**Comment by meganrogge** [comment]
@jooyoungseo, @rperez030 do you have suggestions for the default delays when:  
1. navigating between lines, error (currently 400ms for sound, 3000ms for announcement)
2. navigating between lines, warning (currently 400ms for sound, 3000ms for announcement)
3. navigating between positions, error (currently 100ms for sound, 1000ms for announcement)
4. navigating between positions, warning (currently 100ms for sound, 1000ms for announcement)

@hediet As settings, this could be represented like: 
```
accessibility.signalOptions.errorDebounceTime: {
    soundAtPosition:
    announcementAtPosition:
    soundAtLine:
    announcementAtLine:
}
accessibility.signalOptions.warningDebounceTime: {
    soundAtPosition:
    announcementAtPosition:
    soundAtLine:
    announcementAtLine:
}
```
There are many other ways this could be represented though. WDYT?

**Comment by meganrogge** [comment]
@rperez030 I had misunderstood, here is the current status:

1. navigating between positions, error or warning (currently 100ms for sound, 1000ms for announcement)
2. navigating between lines, any signal (currently 400ms for sound, 3000ms for announcement)

So we should probably have 

```
accessibility.signalOptions.errorDelays: {
    soundAtPosition:
    announcementAtPosition:
    soundAtLine:
    announcementAtLine:
}
accessibility.signalOptions.warningDelays: {
    soundAtPosition:
    announcementAtPosition:
    soundAtLine:
    announcementAtLine:
}
accessibility.signalOptions.generalDelays: {
    soundAtPosition:
    announcementAtPosition:
    soundAtLine:
    announcementAtLine:
}
```
